src/components/activities.js

```diff
@@ -241,6 +241,33 @@ export class ActivityPorts {
   constructor(win) {
     /** @private @const {!web-activities/activity-ports.ActivityPorts} */
     this.activityPorts_ = new WebActivityPorts(win);
+
+    /** @private {?../proto/api_messages.AnalyticsContext} */
+    this.analyticsContext_ = null;
+  }
+
+  /**
+   *
+   * @param {!../proto/api_messages.AnalyticsContext} context
+   */
+  setAnalyticsContext(context) {
+    this.analyticsContext_ = context;
+  }
+  /**
+   * Adds universal arguments
+   * @param {?Object=} opt_args
+   * @return {!Object}
+   * @private
+   */
+  getArgs_(opt_args) {
+    opt_args = opt_args || {};
+    const context = this.analyticsContext_
+      ? this.analyticsContext_.toArray()
+      : null;
+    opt_args = Object.assign(opt_args, {
```

Comment:
Just a heads up, Object.assign mutates the first parameter, so there's no need to write `opt_args = ` here

Reply:
Perhaps this line could be combined with line 263?

Reply:
You could remove line 263, and then have this line be `opt_args = Object.assign({}, opt_args, {'analyticsContext': context})`